container/bin/entrypoint.py

```diff
@@ -0,0 +1,75 @@
+#!/venv/bin/python3
+#
+# SPDX-FileCopyrightText: Red Hat, Inc.
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+import argparse
+import json
+import os
+import subprocess
+
+from ovirt_imageio.client import info
+from ovirt_imageio.client._api import _tmp_dir
+from ovirt_imageio.client._app import setup_signals
+from ovirt_imageio._internal import nbd
+from ovirt_imageio._internal import qemu_nbd
+
+
+TICKET_PATH = "/ticket.json"
+OVIRT_IMAGEIO = "/venv/bin/ovirt-imageio"
+
+
+def _generate_ticket(uuid, socket, size):
+    ticket_info = {}
+    ticket_info["uuid"] = uuid
+    ticket_info["url"] = f"nbd:unix:{socket}"
+    ticket_info["size"] = size
+    ticket_info["timeout"] = 3000
+    ticket_info["inactivity_timeout"] = 300
+    ticket_info["ops"] = ["read", "write"]
+    return ticket_info
+
+
+def gen_ticket(ticket_id, image, socket):
+    img_size = info(image)["virtual-size"]
+    ticket_info = _generate_ticket(ticket_id, socket, img_size)
+    with open(TICKET_PATH, "w+") as f:
+        f.write(json.dumps(ticket_info))
+
+
+def _parse_args():
+    parser = argparse.ArgumentParser(description="ovirt-imageio")
+    parser.add_argument("image", help="Path to image.")
+    parser.add_argument(
+        "-s", "--socket",
+        help="Optional. If set, ticket points to the given socket path. "
+        "Defaults to random temporary path.")
+    parser.add_argument(
+        "--ticket-id",
+        help="Optional. ID of the ticket. Defaults to the image filename.")
+
+    return parser.parse_args()
+
+
+def main():
+    args = _parse_args()
+    # Ticket ID defaults to image filename if not set.
+    ticket_id = (args.ticket_id if args.ticket_id
+                 else os.path.basename(args.image))
+    with _tmp_dir("imageio-") as base:
+        # Use the random base directory if socket is not set.
+        sock_path = args.socket if args.socket else os.path.join(base, "sock")
+
+        gen_ticket(ticket_id, args.image, sock_path)
+        # Setup signals handler to silence backtrace on ctrl+C.
+        setup_signals()
+        sock = nbd.UnixAddress(sock_path)
+        img_format = info(args.image)["format"]
+        # Start services.
+        with qemu_nbd.run(args.image, img_format, sock) as server:
+            subprocess.Popen([OVIRT_IMAGEIO, "-t", TICKET_PATH])
+            server.wait()
```

Comment:
This will wait until qemu-nbd exit, but it will never exit even if iamgeio exit.

We can use subprocess.run() to run imageio and wait until it exist, although it will never terminate:

    with qemu_nbd.run(...):
        subprocess.run([OVIRT_IMAGEIO, "--ticket", TICKET_PATH], check=True)

If imageio terminates, qemu-nbd will also terminate. Also lets always use long
options in code for better readability.